scripts/generate-patch

```diff
@@ -58,16 +58,41 @@ for f in packages/*; do
 				mkdir -p ${f}/charts-original
 				curl -sLf ${url} | tar xvzf - -C ${f}/charts-original --strip ${fields} ${subdirectory} > /dev/null 2>&1
 			fi
+			if [[ -f ${f}/charts/requirements.yaml ]]; then
+				i=0
+				while ! [[ -z $(yq r ${f}/charts/requirements.yaml "dependencies[${i}]") ]]; do
+					subchart=$(yq r ${f}/charts/requirements.yaml "dependencies[${i}].name" | head -n 1)
+					version=$(yq r ${f}/charts/requirements.yaml "dependencies[${i}].version" | head -n 1)
+					helm_repo=$(yq r ${f}/charts/requirements.yaml "dependencies[${i}].repository" | head -n 1)
+					if [[ ${helm_repo} == file://* ]] || [[ ${version} == *'*'* ]]; then
+						# Subchart is tracked locally; patches will not be tracked
+						rm -rf ${f}/charts-original/charts/${subchart}
+						((i=i+1))
+						continue
+					fi
+					# Subchart is tracked in remote Helm repo with a fixed version number; we grab the download URL and get the local chart archive
+					echo "Comparing current changes to ${subchart}-${version} against upstream found at ${helm_repo}"
+					download_url=$(curl -sLf ${helm_repo}index.yaml | cat | yq r - "entries.${subchart}.(version == ${version}).urls[0]")
+					curl -sLf ${download_url} | tar xvzf - -C ${f}/charts-original/charts > /dev/null 2>&1
+					((i=i+1))
+				done
+			fi
 			if [[ -d ${f}/charts ]]; then
 				split_crds=$(cat ${f}/package.yaml | yq r - generateCRDChart.enabled)
 				if [[ "${split_crds}" == "true" ]]; then
 					revert_crd_changes ${f}
 				fi
 				diff -x *.tgz -x *.lock -uNr ${f}/charts-original ${f}/charts > ${f}/$(basename -- ${f}).patch || true
+				if ! [[ -s ${f}/$(basename -- ${f}).patch ]]; then
+					# File is empty; skip creating a patch for it
+					rm ${f}/$(basename -- ${f}).patch
+				fi
 				if [[ "${split_crds}" == "true" ]]; then
 					./scripts/prepare-crds ${f}
 				fi
-				remove_timestamp_from_diff
+				if [[ -f ${f}/$(basename -- ${f}).patch ]]; then
+					remove_timestamp_from_diff
+				fi
```

Comment:
Why is this part after the prepare-crds part ? Can't it go above it ?

Reply:
`remove_timestamp_from_diff` and the prepare-crds part are independent of each other since they touch different files, so it should be fine to place this above. Will update in the next commit!